tests/test_multistore_file.py

```diff
@@ -41,3 +74,170 @@ def test_lock_file_raises_ioerror(self):
                 self.assertTrue(multistore._file.open_and_lock_called)
         finally:
             os.unlink(filename)
+
+    def test_read_only_file_fail_lock(self):
+        credentials = self.create_test_credentials()
+
+        open(FILENAME, 'a+b').close()
+        os.chmod(FILENAME, 0o400)
+
+        store = multistore_file.get_credential_storage(
+            FILENAME,
+            credentials.client_id,
+            credentials.user_agent,
+            ['some-scope', 'some-other-scope'])
+
+        store.put(credentials)
+        if os.name == 'posix':
+            self.assertTrue(store._multistore._read_only)
+        os.chmod(FILENAME, 0o600)
+
+    def test_multistore_no_symbolic_link_files(self):
+        if hasattr(os, 'symlink'):
+            SYMFILENAME = FILENAME + 'sym'
+            os.symlink(FILENAME, SYMFILENAME)
+            store = multistore_file.get_credential_storage(
+                SYMFILENAME,
+                'some_client_id',
+                'user-agent/1.0',
+                ['some-scope', 'some-other-scope'])
+            try:
+                store.get()
+                self.fail('Should have raised an exception.')
```

Comment:
Again hygiene, can you use `self.assertRaises` instead of `self.fail` with the catch?

I think you'll still want to keep the `try/finally` so you can unlink the symlinked file.